buf/registry/module/v1beta1/tag_service.proto

```diff
@@ -92,31 +91,36 @@ message ListTagsResponse {
   repeated Tag tags = 2;
 }
 
-message CreateTagsRequest {
-  // An individual request to create a Tag.
+message SetTagsRequest {
+  // An individual request to set Tags.
   message Value {
-    // The Module to create the Tag under.
-    ModuleRef module_ref = 1 [(buf.validate.field).required = true];
-    // The Tag name.
-    string name = 2 [
+    // The reference to resolve the Commit to which this Tags will be set.
+    //
+    // See the documentation on Ref for resource resolution details.
+    //
+    // The Tags are set depending on the kind of resource that is resolved:
+    //   - If a Module is referenced, Tags are set to the latest released Commit in the module.
+    //   - If a Commit is referenced, Tags are set for that Commit.
+    //   - If a Tag is referenced, Tags are set for the Commit that this Tag references.
+    //   - If a VCSCommit is referenced, Tags are set for the Commit that this VCSCommit references.
+    //   - Is a Branch is referenced, Tags are set for the latest Commit on the Branch.
+    //   - If a Digest is referenced, Tags are set for the most recent Commit with this Digest.
+    ResourceRef resource_ref = 1 [(buf.validate.field).required = true];
+    // The Tags names.
+    repeated string name = 2 [
       (buf.validate.field).required = true,
       (buf.validate.field).string.max_len = 250
     ];
-    // The id of the Commit associated with the Tag.
```

Comment:
How can you create a commit without a commit_id to reference? Every Tag has a commit id.

Reply:
Because the commit would be resolved by the passed `ResourceRef`. This was motivated mainly by Buf Sync in which we have `map[git.Hash][]tags`.